core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/internal/CDOBasedRepository.java

```diff
@@ -235,7 +250,7 @@ public RevisionBranch getParentBranch(String branchPath) {
 		bind(RevisionIndex.class, revisionIndex);
 		// initialize the index
 		index.admin().create();
-		
+		getDelegate().services().registerService(Searcher.class, indexClient.searcher());
```

Comment:
you are right, this is bad to register this to app context.
however i have problem getting the index requests through, as at this point it seems that the searcher is not available at CommitInfoSearchRequest.doExecute(CommitInfoSearchRequest.java:59)
throwing an NPE at me...